Meadowlark-js/backends/meadowlark-opensearch-backend/test/integration/QueryOpensearch.test.ts

```diff
@@ -0,0 +1,304 @@
+
+        expect(result.response).toEqual('QUERY_FAILURE_INVALID_QUERY');
+        expect(result.documents).toHaveLength(0);
+        expect(result.totalCount).toBeUndefined();
+      });
+    });
+
+    describe('when querying with wrong property', () => {
+      it('should return error message', async () => {
+        const result = await queryDocuments(setupQueryRequest({ lastSure: 'Last' }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(0);
+        expect(result.documents).toHaveLength(0);
+      });
+    });
+
+    describe('when querying with non existent data', () => {
+      it('should return empty results', async () => {
+        const result = await queryDocuments(setupQueryRequest({ firstName: 'Last' }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(0);
+        expect(result.documents).toHaveLength(0);
+      });
+    });
+
+    describe('when querying without parameters', () => {
+      it('should return all values', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents[0]).toEqual(student1);
+        expect(result.documents[1]).toEqual(student2);
+      });
+    });
+
+    describe('when querying with valid parameters', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({ firstName: student1.firstName }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(1);
+        expect(result.documents[0]).toEqual(student1);
+      });
+    });
+
+    describe('when querying with limit', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, { limit: 1 }), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents).toHaveLength(1);
+        expect(result.documents[0]).toEqual(student1);
+      });
+    });
+
+    describe('when querying with limit and offset', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, { limit: 1, offset: 1 }), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents).toHaveLength(1);
+        expect(result.documents[0]).toEqual(student2);
+      });
+    });
+
+    describe('when querying with parameters and offset', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(
+          setupQueryRequest({ firstName: student1.firstName }, { limit: 2, offset: 1 }),
```

Comment:
@stephenfuqua this scenario was failing because we need to specify limit when using offset. This is not being limited in the [class](https://github.com/Ed-Fi-Exchange-OSS/Meadowlark/blob/main/Meadowlark-js/backends/meadowlark-opensearch-backend/src/repository/QueryOpensearch.ts#L100). I wonder if we should add a validation to make sure that both are specified.

If not present it will return UNKNOWN_FAILURE

Reply:
We do have validation on that - its just that the validation is in middleware, not back here in the repository.